effect/shared/src/main/scala/scalaz/effect/IOInstances.scala

```diff
@@ -25,4 +26,8 @@ trait IOInstances {
       override def rmap[A, B, T](fab: IO[A, B])(bt: B => T): IO[A, T] = fab.map(bt)
     })
 
+  implicit def isCovariant[E]: IsCovariant[IO[E, ?]] = IsCovariantClass.unsafeForce[IO[E, ?]]
+
+  implicit def isLeftCovariant[A]: IsCovariant[IO[?, A]] = IsCovariantClass.unsafeForce[IO[?, A]]
```

Comment:
Could you add a test `implicitly[IsCovariant[IO[?, Int]]]` to see if it actually resolves?

Reply:
I think we can only have `IsCovariant` for `A`. 

For `E`, we need a newtype. Otherwise, the implicits will be ambiguous.

Also, a newtype can capture `Functor` instances for `E`, as well.

Reply:
I do not quite understand. The code below compiles. Should I add it to the PR?
```scala
class IOCovariance {
  implicitly[IsCovariant[IO[Int, ?]]]
  implicitly[IsCovariant[IO[?, Int]]]
}
```

Reply:
Surprisingly, they are not ambiguous with partial unification (AFAIU it's enabled by default on Scalafiddle): https://scalafiddle.io/sf/KNhiag1/0

Reply:
It picks the one with the rightmost hole, it appears: https://scalafiddle.io/sf/14zY5XS/1

Reply:
Which is to say, I'm not sure how likely you are to be able to _use_ the instance for the left hole: https://scalafiddle.io/sf/3FNT07Y/0